File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithGenericWithArrays.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

import java.util.List;

@KoraApp
public interface AppWithGenericWithArrays {

    @Root
    default Generic<Integer> genericInt() {
        return t -> t;
    }

    @Root
    default Generic<List<Integer>> genericListInt() {
        return t -> t;
    }

    @Root
    default Generic<List<byte[]>> genericListByteArr() {
        return t -> t;
    }

    @Root
    default Generic<List<Object[]>> genericListObjectArr() {
        return t -> t;
    }


    interface Generic<T> {
        T to(T t);
    }
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithGenericWithArrays.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithGenericWithArrays {
    @Root
    fun genericInt(): Generic<Int> {
        return object : Generic<Int> {
            override fun to(t: Int): Int {
                return t
            }
        }
    }

    @Root
    fun genericListInt(): Generic<List<Int>> {
        return object : Generic<List<Int>> {
            override fun to(t: List<Int>): List<Int> {
                return t
            }
        }
    }

    @Root
    fun genericListByteArr(): Generic<List<ByteArray>> {
        return object : Generic<List<ByteArray>> {
            override fun to(t: List<ByteArray>): List<ByteArray> {
                return t
            }
        }
    }

    @Root
    fun genericListObjectArr(): Generic<List<Array<Any>>> {
        return object : Generic<List<Array<Any>>> {
            override fun to(t: List<Array<Any>>): List<Array<Any>> {
                return t
            }
        }
    }

    interface Generic<T> {
        fun to(t: T): T
    }
}
